featurize_pdbbind_cluster_job.py

```diff
@@ -86,6 +96,63 @@ def featurize_job(pdb_directories, pickle_out):
   with open(pickle_out, "wb") as f:
     pickle.dump(feature_vectors, f)
 
+def featurize_3d_grid(pdb_directories, pickle_out, box_width, voxel_width, tmp_dir):
+  feature_vectors = {}
+  for count, dir in enumerate(pdb_directories):
+    print "\nprocessing %d-th pdb %s" % (count, dir)
+
+    print "About to extract ligand and protein input files"
+    ligand_pdb, ligand_pdbqt = None, None
+    protein_pdb, protein_pdbqt = None, None
+    for f in os.listdir(dir):
+      if re.search("_ligand_hyd.pdb$", f):
+        ligand_pdb = f
+      elif re.search("_ligand_hyd.pdbqt$", f):
+        ligand_pdbqt = f
+      elif re.search("_protein_hyd.pdb$", f):
+        protein_pdb = f
+      elif re.search("_protein_hyd.pdbqt$", f):
+        protein_pdbqt = f
+
+    print "Extracted Input Files:"
+    print (ligand_pdb, ligand_pdbqt, protein_pdb, protein_pdbqt)
+    if (not ligand_pdb or not ligand_pdbqt or not protein_pdb or not
+        protein_pdbqt):
+        raise ValueError("Required files not present for %s" % dir)
+
+    ligand_pdb_path = os.path.join(dir, ligand_pdb)
+    ligand_pdbqt_path = os.path.join(dir, ligand_pdbqt)
+    protein_pdb_path = os.path.join(dir, protein_pdb)
+    protein_pdbqt_path = os.path.join(dir, protein_pdbqt)
+
+    tmp_dir = os.path.join(tmp_dir, protein_pdb)
+    if not os.path.exists(tmp_dir): os.makedirs(tmp_dir)
+    system_pdb = os.path.join(tmp_dir, "system.pdb")
+    box_pdb = os.path.join(tmp_dir, "box.pdb")
+    box_pickle = os.path.join(tmp_dir, "box.pickle")
+    grid_pickle = os.path.join(tmp_dir, "grid.pickle")
+
+    p = PDBTransformer()
+    p.transform(protein_pdb_path, protein_pdbqt_path, ligand_pdb_path, ligand_pdbqt_path, system_pdb, box_pdb, box_pickle, box_x = box_width, box_y = box_width, box_z = box_width)
+
+    g = GridGenerator()
+    grid = g.transform(box_pickle, box_width, box_width, box_width, voxel_width, grid_pickle, num_features=3)
+
+    feature_vectors[dir] = grid
```

Comment:
This should be

```
feature_vectors[dir] = (grid, smiles, seq)
```

where smiles is the smiles string of the ligand and seq is the sequence of the protein. Check the original featurize_pdbbind_cluster_job.py file to see how these are computed.